src/main/java/org/broadinstitute/dsm/model/Filter.java

```diff
@@ -271,6 +259,48 @@ else if (filter.getFilter1() != null && filter.getFilter2().getValue() != null &
         return finalQuery;
     }
 
+    private static String buildJsonExtract(Filter filter, DBElement dbElement) {
+        String query;
+        String finalQuery;
+        String jsonExtract = "JSON_EXTRACT ( ";
+        query = AND +
+                jsonExtract + filter.getParentName() + DBConstants.ALIAS_DELIMITER + dbElement.getColumnName() + " , '$." + filter.getFilter2().getName() + "' ) ";
+        if (filter.isEmpty()) {
+            finalQuery = query + IS_NULL + " ";
+        }
+        else if (filter.isNotEmpty()) {
+            finalQuery = query + IS_NOT_NULL + " ";
+        }
+        else {
+            String notNullQuery = AND + filter.getParentName() + DBConstants.ALIAS_DELIMITER + dbElement.getColumnName() + IS_NOT_NULL;
+            if (filter.getFilter1() != null && filter.getFilter1().getValue() != null && StringUtils.isNotBlank(String.valueOf(
+                    filter.getFilter1().getValue()))) {
+                query = AND + jsonExtract + filter.getParentName() + DBConstants.ALIAS_DELIMITER + dbElement.getColumnName() + " , '$." + filter.getFilter2().getName() + "' ) ";
+                if (BaseFilterParticipantList.isDateRange(filter)) {
+                    String moreThan = generateDateComparisonSql(filter, dbElement, LARGER_EQUALS, filter.getFilter1().getValue(), false);
+                    String lessThan = generateDateComparisonSql(filter, dbElement, SMALLER_EQUALS, filter.getFilter2().getValue(), true);
+                    int moreThanIndex = moreThan.indexOf(Filter.LARGER_EQUALS_TRIMMED);
+                    int lessThanIndex = lessThan.indexOf(Filter.SMALLER_EQUALS_TRIMMED);
+                    moreThan = moreThan.substring(moreThanIndex);
+                    lessThan = lessThan.substring(lessThanIndex);
+                    query += moreThan + query + lessThan;
+                } else if (filter.isExactMatch()) {
+                    query += EQUALS + "'#'";
+                    query = query.replaceAll("#", String.valueOf(filter.getFilter1().getValue()));
+                } else {
+                    query += " " + LIKE + " '%#%'";
+                    query = query.replaceAll("#", String.valueOf(filter.getFilter1().getValue()));
```

Comment:
possible sql injection?

Reply:
whole Filter class can have possible sql injection.

Reply:
can you come up with an idea how to protect us from it?

Reply:
our ES filtering is built depend on this class, if we would choose the way to eliminate this class fully we would have to change the whole logic of filtering, which would involve tons of changes on frontend side as well, that's why we decided to use already existing logic of frontend-backend for filtering, if we consider that this class is only used for filtering and we don't filter internal database anymore and we build ES queries according to this class, we can say that we won't have sql injection, but we need to keep in mind that we should try to avoid that class to filter something in database.

Reply:
I am totally fine to keep that class and not change the logic, I was just wondering if we could add some checks on the value that would prevent sql injection…